feathr_project/feathr/api/api.py

```diff
@@ -0,0 +1,198 @@
+import os, json
+import logging
+from typing import List, Optional
+from fastapi import FastAPI, HTTPException, Response, status
+from pyapacheatlas.core import (AtlasException)
+from pydantic import BaseModel
+from opencensus.ext.azure.log_exporter import AzureLogHandler
+from azure.core.credentials import AzureKeyCredential
+from azure.core.exceptions import HttpResponseError, ServiceRequestError, ResourceNotFoundError
+from feathr.protobuf.featureValue_pb2 import FeatureValue
+from feathr._feature_registry import _FeatureRegistry
+from feathr._file_utils import write_to_file
+from feathr._synapse_submission import _FeathrSynapseJobLauncher
+from feathr._databricks_submission import _FeathrDatabricksJobLauncher
+from feathr.query_feature_list import FeatureQuery
+from feathr.settings import ObservationSettings
+from feathr.constants import *
+from feathr._feature_registry import _FeatureRegistry
+
+
+
+app = FastAPI()
+
+# Log Level
+log_level = os.getenv("logLevel", "INFO")
+
+logger = logging.getLogger(__name__)
+handler = logging.StreamHandler()
+# Set the application insights connection string to enable the logger to write to Application Insight's directly
+app_insights_connection_string = os.getenv("APPLICATIONINSIGHTS_CONNECTION_STRING")
+formatter = logging.Formatter("[%(asctime)s] [%(name)s:%(lineno)s - %(funcName)5s()] %(levelname)s - %(message)s")
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+if app_insights_connection_string:
+    azure_handler = AzureLogHandler(connection_string=app_insights_connection_string)
+    logger.addHandler(azure_handler)
+else:
+    logger.warning("APPLICATIONINSIGHTS_CONNECTION_STRING is not set - will NOT log to AppInsights!!")
+logger.setLevel(log_level)
+logger.info("starting %s", __file__)
+
+"""
+This is to enable Authentication for API and not keep it wide open.
+You can set the AppServiceKey variable as environment variable
+and make sure to pass the variable as query parameter when you access the API.
+eg - <apiserver>/projects/<project_name>/features?code=<your_api_code>
+"""
+appServiceKey = os.getenv("AppServiceKey")
```

Comment:
Why is this not using EnvUtil? @xiaoyongzhu @xiaoyongzhumsft

Reply:
I purposefully didn't use envutil as it requires the config file to be passed around and it becomes a challenge for the API app when we have to dockerize it, the file either needs to be packaged together with app or we need to change the logic in EnvUtil to accept wasb:// location as input_path not just local_file_path. Currently we can easily set environment variables for Azure webapp, while deploying the app.